src/metadataGeneration/typeResolver.ts

```diff
@@ -254,6 +254,29 @@ export class TypeResolver {
         return stringMetaType;
       }
 
+      const type = this.current.typeChecker.getTypeFromTypeNode(this.typeNode);
```

Comment:
I didn't see this initially, resolving based on toJSON() only makes sense when we are at the point of resolving a Model right around [here](https://github.com/lukeautry/tsoa/blob/e151d2ec01b741c7bc0952aeb71b4025f9278ef2/src/metadataGeneration/typeResolver.ts#L480)